pkg/apis/alerting/v1alpha/alerting.proto

```diff
@@ -0,0 +1,206 @@
+import "google/api/annotations.proto";
+import "google/rpc/status.proto";
+import "github.com/rancher/opni/pkg/apis/core/v1/core.proto";
+
+service Alerting {
+  // ------- Trigger Alerts -------
+  // TODO
+  rpc TriggerAlerts(TriggerAlertsRequest) returns (TriggerAlertsResponse) {
+    option (google.api.http) = {
+      post: "/alerts/v1alpha/trigger"
+      body: "*"
+    };
+  }
+
+  // ------- CRUD for alerting events log -------
+  rpc CreateAlertLog(core.AlertLog) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/alerts/v1alpha/log"
+      body: "*"
+    }; 
+  }
+
+  // id is the unix epoch timestamp of the alert 
+  rpc GetAlertLog(core.Reference) returns (core.AlertLog) {
+    option (google.api.http) = {
+      get: "/alerts/v1alpha/events/{id}"
+    };
+  }
+
+  rpc ListAlertLogs(ListAlertLogRequest) returns (core.AlertLogList) {
+    option (google.api.http) = {
+      get: "/alerts/v1alpha/events"
+    };
+  }
+
+  rpc UpdateAlertLog(UpdateAlertLogRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      put: "/alerts/v1alpha/events"
+      body: "*"
+    };
+  }
+
+  rpc DeleteAlertLog(core.Reference) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/alerts/v1alpha/events"
+    };
+  }
+
+  // -------- CRUD for alerting conditions -------
+  //TODO 
+  rpc CreateAlertCondition(AlertCondition) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/alerts/v1alpha/conditions"
+      body: "*"
+    };
+  }
+
+  rpc GetAlertCondition(core.Reference) returns (AlertCondition) {
+    option (google.api.http) = {
+      get: "/alerts/v1alpha/conditions/{id}"
+    };
+  }
+
+  rpc ListAlertConditions(ListAlertConditionRequest) returns (AlertConditionList) {
+    option (google.api.http) = {
+      get: "/alerts/v1alpha/conditions"
+    };
+  }
+
+  rpc UpdateAlertCondition(UpdateAlertConditionRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      put: "/alerts/v1alpha/conditions"
+      body: "*"
+    };
+  }
+
+  rpc DeleteAlertCondition(core.Reference) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      delete: "/alerts/v1alpha/conditions"
+    };
+  }
+
+  rpc PreviewAlertCondition(PreviewAlertConditionRequest) returns (PreviewAlertConditionResponse) {
+    option (google.api.http) = {
+      post: "/alerts/v1alpha/conditions/preview"
+      body: "*"
+    };
+  }
+
+  // -------- CRUD for alerting endpoints ----
+  // TODO 
+  rpc CreateAlertEndpoint(AlertEndpoint) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      post: "/alerts/v1alpha/endpoints"
+      body: "*"
+    };
+  }
+
+  rpc GetAlertEndpoint(core.Reference) returns (AlertEndpoint) {
+    option (google.api.http) = {
+      get: "/alerts/v1alpha/endpoints/{id}"
+    };
+  }
+
+  rpc ListAlertEndpoints(ListAlertEndpointsRequest) returns (AlertEndpointList) {
```

Comment:
You should use `google.protobuf.Empty` for empty structs

Reply:
Sorry, this one wasn't clear the `ListAlertEndpointsRequest` will contain something like filter/group by options